midealocal/security.py

```
@@ -245,6 +245,19 @@ def sign(self, url: str, data: dict[str, Any] | str, random: str) -> str:  # noq
         sha.update((urlparse(url).path + payload + self._login_key).encode("ascii"))
         return sha.hexdigest()
 
+    def decrypt_appliance_lua(self, data: str) -> str:
+        """Decrypt a legacy (mapp.appsmb.com) appliance lua file.
+
+        Unlike the Meiju/SmartHome clouds, the legacy backend serves lua files
+        as a hex-encoded AES-128-ECB blob keyed by ``md5(app_key)[:16]``, where
+        ``app_key`` is this security object's login key.
+        """
+        key = md5(self._login_key.encode("ascii")).hexdigest()[:16].encode("ascii")
```

Ruff (0.16.0)

[error] Probable use of insecure hash functions in `hashlib`: `md5`

(S324)